### DISCUSSION TITLE
My Roslyn analyzer not picked up by Visual Studio 2022 - how to debug?

### DISCUSSION THREAD

**Discussion by MartyIX** [dotnet/roslyn#56608] [Q&A] [answered]
I have an analyzer and it works fine when it is run via VSIX but when I add it to a consumer project via package reference, it does not work (nothing really happens).

I can see my analyzer in the consumer project in Visual Studio like this:

![image](https://user-images.githubusercontent.com/203266/134344053-764f5784-ca00-4ed4-b54e-2d71c6bb5eb3.png)

The bottom two analyzer+code fix are mine but they show no rules (no arrows as the other analyzers have). So I guess they are not picked up correctly but I don't really know what affects that or if there is some useful log to understand it more.

The reference in the consumer project looks like this:

```xml
<PackageReference Include="Project.ScriptAnalyzer" Version="1.0.0">
  <PrivateAssets>all</PrivateAssets>
  <IncludeAssets>runtime; build; native; contentfiles; analyzers; buildtransitive</IncludeAssets>
</PackageReference>
```

I have used the roslyn analyzer template to create my analyzer from scratch and built `ScriptAnalyzer.Package` to get a nuget file which I then copied to a local nuget repository folder from which the consumer project gets the analyzer.

Any idea what to try to make it work?

**Comment by jmarolf** [comment, marked as answer]
Do you have dependencies on things other than the compiler? This is normally caused by either
1. your analyzer accidentally referencing types from your code fix
2. You adding a nuget dependency and not having it included in the nuget package

**Reply by MartyIX** [reply]
So it's my mistake. I did rename template projects and I did not delete:

```xml
    <!-- Avoid ID conflicts with the package project. -->
    <PackageId>*$(MSBuildProjectFile)*</PackageId>
```

from the main analyzer project. I'm not entirely sure the inner workings but what I did was basically I started with the roslyn template and gradually applied my changes to find out the cuplrit. Too bad that a log file cannot somehow show it to me but it works and I'm happy :)
